security/WFLY-14922_Remove_PicketLink_Subsystem.adoc

```diff
@@ -0,0 +1,125 @@
+can be used instead.
+
+This task is focussed on the removal of this subsystem from WildFly and the related migration path
+for configuring SSO with SAML for WildFly.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14922[WFLY-14922 - Remove picketlink extension and subsystems]
+* https://issues.redhat.com/browse/WFLY-15076[WFLY-15076 - Add documentation on how to configure SSO with SAML for WildFly using the Keycloak adapter]
+* https://issues.redhat.com/browse/KEYCLOAK-18924[KEYCLOAK-18924 - The org.picketbox module dependency should be removed from the SAML Elytron adapter module.xml file]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1750[EAP7-1750 - Remove PicketLink Subsystem]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [x] QE
+
+As a removal RFE, this does not require new feature test development. However both teams will need
+to review existing testsuites for tests dependent on the PicketLink subsystem. Identified tests will
+either need to be deleted or they will need to be converted to use the new approach.
+
+=== Affected Projects or Components
+
+* WildFly
+* Keycloak
+
+=== Other Interested Projects
+
+* [x] External Testsuites
+
+* [x] Quickstarts
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+Since this is a removal task, this section will describe the steps that will be needed as opposed to the
+set of requirements.
+
+=== Hard Requirements
+
+* The PicketLink subsystem needs to be converted to a skeleton subsystem. (It needs to be a skeleton subsystem
+instead of being removed completely so that older domains can still be managed.)
+
+* Any affected tests will need to be removed from WildFly.
+
+* Any documentation that references PicketLink will need to be removed. The main section in the WildFly documentation
+that is affected is the "WS-Trust and STS section".
+
+* The `org.keycloak.keycloak-saml-wildfly-elytron-adapter` module that's installed by the Keycloak SAML Elytron
+adapter needs to be updated to remove the `org.picketbox` module from its dependencies (the `org.picketbox` dependency
+does not appear to be used, it just needs to be removed).
+
+=== Nice-to-Have Requirements
+
+N/A
+
+=== Non-Requirements
+
+* The Keycloak SAML Elytron adapter does not ship with WildFly. It needs to be installed separately by the
+user. Thus, migrating from SSO with SAML using PicketLink to SSO with SAML via the Keycloak adapter won't be
+something that's handled by the server migration tool. The user will need to install the Keycloak adapter on
```

Comment:
I agree that this is out of scope of this RFE. However, there was a discussion that the migration via migration tool could be possible.

Reply:
Is https://issues.redhat.com/browse/WFLY-15623 part of this RFE?

Reply:
Yes, WFLY-15623 will be part of this. I'll need to update the analysis to reflect that.

Reply:
I've added more information about migration to the document.